tests/main/remove-core/task.yaml

```diff
@@ -0,0 +1,32 @@
+summary: Check that we can remove core snap on classic systems.
+
+details: |
+    This test checks that we can remove core snap on classic systems, as long as
+    snapd is installed as a snap. Classic systems do not have 'base' set in
+    their model. On UC systems, that would be interpreted as 'core' being the
+    base. On classic (non-hybrid) systems, that should be interpreted as there
+    not being a base required.
+
+    If snapd is not installed as a snap, then we can't remove the core snap
+    since it might be providing snapd.
+
+systems: [ubuntu-22.04-64]
+
+execute: |
+    # we should not be able to remove the core snap, since the snapd snap is not
+    # installed.
+    not snap remove core
```

Comment:
I'm surprised that we are in this state, as it shouldn't be a common state nowadays, but ok